packages/recorder/src/messageBus.ts

```diff
@@ -0,0 +1,9 @@
+import { EventEmitter } from 'eventemitter3';
+
+import { VideoQueueItem } from './types';
+
+interface EventSpec {
+  'video-written': (data: VideoQueueItem) => void;
+}
+
+export class ManagerMessageBus extends EventEmitter<EventSpec> {}
```

Comment:
can't remember exactly but it might need to be something like EventEmitter<EventNames, EventArgs>

Reply:
where type EventNames = 'event1' | 'event2';